src/main/java/com/coravy/hudson/plugins/github/GithubLinkAnnotator.java

```diff
@@ -43,8 +43,14 @@ void annotate(final GithubUrl url, final MarkupText text, final Entry change) {
         if (change instanceof GitChangeSet) {
             GitChangeSet cs = (GitChangeSet) change;
             final String id = cs.getId();
-            text.wrapBy("", " (<a href='" + url.commitId(id)
-                        + "'>commit: " + id.substring(0, Math.min(id.length(), 7)) + "</a>)");
+            final String innerText = "commit: " + id.substring(0, Math.min(id.length(), 7));
+            final String startHREF = "<a href='" + url.commitId(id) + "'>";
+            final String endHREF = "</a>":
+            text.wrapBy("", " ("
+                        + startHREF
+                        + innerText
+                        + endHREF
+                        + ")");
```

Comment:
@lanwen was this the type of formatting you were envisioning? It seems to clarify the subcomponents of the markup, but I'm not sure that is what you were expecting?